src/System.Management.Automation/engine/TypeTable.cs

```diff
@@ -4491,11 +4489,6 @@ private bool ProcessIsBuiltIn(string filePath, ConcurrentBag<string> errors, out
                 ProcessTypeData(filePath, errors, Types_Ps1Xml.Get());
                 result = true;
             }
-            else if (string.Equals(Path.Combine(psHome, "typesv3.ps1xml"), filePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
This method is used by `public TypeTable(IEnumerable<string> typeFiles)`. This is a public API and this change is potentially a breaking change if the `typeFiles` passed to that public constructor contains `types3.ps1xml`.

Reply:
I added new commit to partially address the breaking. 

After that I hope the breaking change will be in gray area. 
The only scenario that we cannot fix is ​​to load 'typesv3.ps1xml' file without 'types.ps1xml'. This scenario is incredible.